### PR TITLE
add file header decoration comments if configured

### LINKED ISSUE #2086: FileHeaderCodeFixProvider should provide same file header fix as in Stylecop classic with R#
The current fix result looks like this:

```
// <copyright file="Xxx.cs" company="...">
// Copyright ....
// </copyright>
```

StyleCop classic fix also added some "decoration" lines like this:

```
// -----------------------------------------------------------------------
// <copyright file="Xxx.cs" company="...">
// Copyright ....
// </copyright>
// -----------------------------------------------------------------------
```

I'd like to have that decoration lines back as our whole code base has this header style.

One proposal would be to have there an additional setting in stylecop.json, e.g. like this:

```
  "settings": {
    "documentationRules": {
      "companyName": "...",
      "headerDecoration": "-----------------------------------------------------------------------",
    }
  }
```

I have already tested it, I'll attach the pull request (no test written yet).

### PR DESCRIPTION
**Pull request by ThomasZitzler**
Fixes #2086

I have written no test yet, I first want to know if you accept this.

### CODE REVIEW

**Comment by ThomasZitzler** [comment]
Added a test now. Anything else I have to do?

**Comment by vweijsters** [comment]
As the settings file format is updated, the following files should be updated as well:
- `documentation/Configuration.md`
- `StyleCop.Analyzers/StyleCop.Analyzers/Settings/stylecop.schema.json`

**Comment by vweijsters** [comment]
The code looks good to me.

**Comment by ThomasZitzler** [comment]
I discovered a bug that the fix for SA1638 creates duplicate decorations. I fixed it with the last commit. Is there anything else to do?

File: StyleCop.Analyzers/StyleCop.Analyzers/Settings/stylecop.schema.json [hunk 1]
```diff
@@ -194,6 +194,11 @@
               "description": "Determines whether the file header should be wrapped in the StyleCop-standard XML structure.",
               "default": true
             },
+			"headerDecoration": {
+              "type": "boolean",
```
Comment (pdelvo) [inline StyleCop.Analyzers/StyleCop.Analyzers/Settings/stylecop.schema.json]: Im not familiar with json schemas, but shouldn't this be string?

**Comment by pdelvo** [comment]
Other then the minor comment this looks good to me. @sharwell should be back next week to merge this if he does not have other comments on this..

File: StyleCop.Analyzers/StyleCop.Analyzers/Settings/stylecop.schema.json (hunk 1 above)
Comment (ThomasZitzler) [inline StyleCop.Analyzers/StyleCop.Analyzers/Settings/stylecop.schema.json]: You are totally right, probably copy+paste :-)

**Pull request merged**